### PR TITLE
INTLY-6603 Create end-to-end document covering Tower release process

### PR DESCRIPTION
**Pull request by Boomatang**
## Additional Information
**JIRA** https://issues.redhat.com/browse/INTLY-6603

- Git template updated
- SOP updated for removing variables from configs

## Verification Steps
As the verifier of the PR the following process should be done:

## Checklist

- [ ] Updated [CHANGELOG](https://github.com/integr8ly/tower_dummy_credentials/blob/master/CHANGELOG.md)

### CODE REVIEW

**Comment by Boomatang** [comment]
@pmccarthy @StevenTobin I can not added reviewers for this repo. Do you mind having a look.

File: README.md [hunk 1]
```diff
@@ -66,6 +66,24 @@ The following steps outline the process of adding new variables to the project.
 
         new_variable: {{ new_variable }}
 
+### Removing old variables
+
+Removed variables will be documented in the [changelog](CHANGELOG.md) for each release. Review the [changelog](CHANGELOG.md) to ensure that all on wanted variables are removed from the configuration files.
+
+#### Encrypted
+
+1. Remove the variable form the `CREDENTIAL_CONFIG.yml` file.
```
Comment (pmccarthy) [inline README.md]: Small typo. Should be `from` rather than `form`

File: README.md [hunk 2]
```diff
@@ -66,6 +66,24 @@ The following steps outline the process of adding new variables to the project.
...
+
+2. Insure the variable has been removed from the `bootstrap_credentials.yml`. This should have been done as part of the update.
```
Comment (pmccarthy) [inline README.md]: Typo here,  `Ensure` rather than `Insure`

Comment (pmccarthy) [inline README.md]: "This should have been done as part of the update." - not quite sure what is being referred to here? If a variable already exists it must be in the `bootstrap_credentials.yml` file?

File: README.md [hunk 3]
```diff
@@ -66,6 +66,24 @@ The following steps outline the process of adding new variables to the project.
...
+
+3. Insure the variable has been removed from all templates. From the repo's root grep for instances of the variable. This should be an empty return. `grep -rw <old_variable> *`
```
Comment (pmccarthy) [inline README.md]: Might be easier to copy/paste the `grep` command if it was on a new line. Would also be good to keep some consistency for code blocks e.g. new line and indent command. See other sections of the README for examples

**Review by pmccarthy — CHANGES_REQUESTED** [review]
@Boomatang left some comments here, all very minor to be fair. Overall it looks good though

File: README.md (hunk 2 above)
Comment (Boomatang) [reply README.md]: I have reworded this a small bit, hopefully its more understandable.

**Comment by Boomatang** [comment]
Section added for reporting changes back.

File: README.md [hunk 4]
```diff
@@ -4,6 +4,23 @@ Dummy credentials repository used to generate organisation specific credential r
 
 **NOTE:** This repository serves as the inventory source for the [Ansible Tower Configuration](https://github.com/integr8ly/ansible-tower-configuration) repository.
 
+## Reporting changes back
```
Comment (pmccarthy) [inline README.md]: Would you mind moving this to the end of the README please? I think it makes sense to leave Getting Started as the first section here

File: README.md [hunk 5]
```diff
@@ -66,6 +83,32 @@ The following steps outline the process of adding new variables to the project.
 
         new_variable: {{ new_variable }}
 
+### Removing old variables
+
+Removed variables will be documented in the [changelog](CHANGELOG.md) for each release. Review the [changelog](CHANGELOG.md) to ensure that all on wanted variables are removed from the configuration files.
```
Comment (pmccarthy) [inline README.md]: Small typo, should be `unwanted` rather than `on wanted`?

File: README.md [hunk 6]
```diff
@@ -66,6 +83,32 @@ The following steps outline the process of adding new variables to the project.
...
+
+#### Encrypted
+
+1. Remove the variable from the `CREDENTIAL_CONFIG.yml` file.
+
+2. Ensure the variable has been removed from the `bootstrap_credentials.yml`.
```
Comment (pmccarthy) [inline README.md]: Should this read "from the `bootstrap_credentials.yml` file"?

**Review by pmccarthy — CHANGES_REQUESTED** [review]
@Boomatang Added a couple more comments, all very minor. Once fixed we should be good to merge back 👍

File: README.md (hunk 4 above)
Comment (Boomatang) [reply README.md]: I see where you are coming from but I disagree. You want people to be reminded they have to report changes back. I feel that if this is moved to the bottom some people may read and not remember to push the changes back. If its at the start it puts the idea of reporting changes back in their mind even if they only read the title of the section.

I will gladly change this if you wise, I just wanted you to understand my point of view.

To be clear, Do you want me to move this to the bottom?

Comment (pmccarthy) [reply README.md]: @Boomatang Fully understood and I agree with your points here. Perhaps as a compromise we could have a **Prerequisites** heading at the top of the README and move **Reporting changes back** as a subheading under this section? That way the flow would be as follows (which to me makes more sense):

- Prerequisites
-- Reporting changes back
- Getting Started
- .........

Comment (Boomatang) [reply README.md]: Yap. That makes sense to me. I will move it under Prerequisites.

**Review by pmccarthy — APPROVED** [review]
@Boomatang Looks good 👍

**Pull request merged**